File: app/src/main/java/com/example/practicals/java_practicals/Inheritence.java
package com.example.practicals.java_practicals;

class Student {
    String name;
    public void baseFun() {
        System.out.println("parent function");
    }
}

class Batch extends Student {
    public void show() {
        System.out.println("name : " + name);
    }
}

public class Inheritence {
    public static void main(String[] args) {
        Batch batch = new Batch();
        batch.name = "mansi";
        batch.show();
        batch.baseFun();

    }
}

File: app/src/main/java/com/example/practicals/kotlin_practicals/Inheritence.kt
package com.example.practicals.kotlin_practicals

open class parent {
    val name = "mansi"
}

class Inheritence : parent() {
    fun printName() {
        println("name :" + name)
    }
}

fun main() {
    val c1 = Inheritence()
    c1.printName()
}